src/main/player/Player.java

```diff
@@ -11,14 +11,57 @@
     protected int up, left, right; // direction for player
     protected int leftBorder, rightBorder;
     public boolean ifStart;
-
+    
     Player(int x, int y) {
         super(x, y);
         this.ifJump = false;
         this.jumpSpeed = -10;
         this.ifStart = true;
     }
 
+    public void MoveByBall(int ball_x, int ball_y, float speed_x, float speed_y){
```

Comment:
* I think you can just follow ball's x coordinate, no need to do such complicated calculations, but I'm not sure how effective it is, you can try and see
* you can rewrite the `move` function by send parameters with `ballX=null`, `ballY=null`, and judge if have `ballX`, if have then follow it, this is just my idea, do whatever u like

Reply:
- I have tried following ball's coordinates before doing calculations, but it didn't work well since the player moved slower than the ball. 
- I still keep the `moveByBall` function since I think it doesn't work like `move` function. The function of `moveByBall` is more like replacing `keyReleased` and `keyPressed`.